geopandas/io/arrow.py

```diff
@@ -247,16 +259,25 @@ def _validate_metadata(metadata):
             )
 
 
-def _geopandas_to_arrow(df, index=None, schema_version=None):
+def _geopandas_to_arrow(df, index=None, schema_version=None, bbox_column_name=None):
     """
     Helper function with main, shared logic for to_parquet/to_feather.
     """
-    from pyarrow import Table
+    from pyarrow import Table, StructArray
+
+    if bbox_column_name:
+        bounds = df.bounds
+        df[bbox_column_name] = StructArray.from_arrays(
```

Comment:
I think you will have to add this column directly to the pyarrow Table (so after `table = Table.from_pandas(df, preserve_index=index)` below). Otherwise this line will still convert the pyarrow struct array to a numpy object dtype array of python dicts, so not avoiding this unnecessary costly conversion